[flang][Driver] Enable support for -mmacos-version-min=

tblah opened the pull request:
So far as I can tell this option is driver-only so we can just re-use what already exists for clang. I've added a unit test based on clang's unit test to demonstrate that the option is handled.

Still TODO is to ensure that flang-rt is built with the same macos minimum version as compiler-rt. At the moment, setting the flang minimum version to older than the macos version on which flang was built will lead to link warnings because flangrt is built for version of macos on which flang was built rather than the oldest supported version (as compiler-rt is).

tblah committed:
[flang][Driver] Enable support for -mmacos-version-min=

So far as I can tell this option is driver-only so we can just re-use
what already exists for clang. I've added a unit test based on clang's
unit test to demonstrate that the option is handled.

tblah commented:
The failing tests are from compiler-rt. I don't think they are related to this patch.

tarunprabhu reviewed (commented):
Thanks Tom. Since this has been enabled for `fc1` as well, could you add a test to ensure that it is accepted by `fc1` as well. Just one should be fine. We don't have to check everything in both cases.

tblah committed:
Remove -fc1 option

There is a clang -cc1 option but so far as I can tell it is unused.

tblah commented:
Nice catch. I added the -fc1 option because there is a -cc1 option. But it doesn't seem to actually be used at all by the frontend driver so I just removed the -fc1 option. I left the clang option just in case.

tarunprabhu reviewed (approved):
LGTM. Thanks.
